lib/block_view_association.go

```diff
@@ -639,11 +639,33 @@ func _isValidUserAssociationQuery(associationQuery *UserAssociationQuery) error
 		len(associationQuery.AssociationTypePrefix) == 0 &&
 		len(associationQuery.AssociationValue) == 0 &&
 		len(associationQuery.AssociationValuePrefix) == 0 {
-		return errors.New("invalid query params")
+		return errors.New("invalid query params: none provided")
 	}
-	if (len(associationQuery.AssociationType) > 0 && len(associationQuery.AssociationTypePrefix) > 0) ||
-		(len(associationQuery.AssociationValue) > 0 && len(associationQuery.AssociationValuePrefix) > 0) {
-		return errors.New("invalid query params")
+	if len(associationQuery.AssociationType) > 0 && len(associationQuery.AssociationTypePrefix) > 0 {
+		return errors.New("invalid query params: both AssociationType and AssociationTypePrefix provided")
+	}
+	if len(associationQuery.AssociationValue) > 0 && len(associationQuery.AssociationValuePrefix) > 0 {
+		return errors.New("invalid query params: both AssociationValue and AssociationValuePrefix provided")
+	}
+	if associationQuery.Limit < 0 {
```

Comment:
do we want to enforce an upper bound on limit here?

Reply:
I was thinking I wouldn't in core. But would in backend. That makes it easier to change. I was thinking by default the Limit would be 100. And the max would possibly be 100, 500, or 1k depending on performance.